common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
@dsjen Agreed, it would probably make sense to give them access as well.  However, I currently don't see a sane way of achieving this.

We could meld the access checking functions in `courseware.access` into the corresponding functions into `student.auth`, or at least merge part of the functionality.  This way, we could improve the current situation of having overlapping but slightly different implementations of these checks, and additionally make the proper checks available outside of the LMS.  This seems a worthwhile project to me, but one that is outside the scope of this PR. :)